pfio/ArrayReference.F90

```diff
@@ -25,6 +25,7 @@ module pFIO_ArrayReferenceMod
    end type ArrayReference
 
    interface ArrayReference
+      module procedure new_ArrayReference_from_param
```

Comment:
While we are here,  `module` is now superfluous for these.  Please remove in next pass.